cpp/serve/engine.cc

```diff
@@ -164,15 +168,18 @@ class EngineImpl : public Engine {
   Optional<PackedFunc> request_stream_callback_;
   // Engine actions.
   Array<EngineAction> actions_;
+  // Event logger.
+  Optional<EventLogger> logger_;
```

Comment:
Terminology, i think it is better to call it a tracer , see https://en.wikipedia.org/wiki/Tracing_(software)